core/src/engine/task/task_queue.cpp

```diff
@@ -37,15 +41,18 @@ boost::intrusive_ptr<impl::TaskContext> TaskQueue::PopBlocking() {
 
 void TaskQueue::StopProcessing() { DoPush(nullptr); }
 
+std::size_t TaskQueue::GetSize() const noexcept { return queue_.size_approx(); }
+
 std::size_t TaskQueue::GetSizeApproximate() const noexcept {
-  return queue_.size_approx();
+  return queue_size_cached_.load(std::memory_order_acquire);
 }
 
 void TaskQueue::DoPush(impl::TaskContext* context) {
   // This piece of code is copy-pasted from
   // moodycamel::BlockingConcurrentQueue::enqueue
   queue_.enqueue(context);
   queue_semaphore_.signal();
+  UpdateQueueSize();
```

Comment:
Can we move it to `task_processor.cpp`, as well as `queue_size_cached_`?
It seems like something independent of the task queue implementation

Reply:
Discussed in PM, will remove `queue_size_cache_` altogether